### ISSUE TITLE
Postgres is not actually used by Keycloak

### ISSUE THREAD

**Issue by andreygoran** [brakmic/Keycloak_on_Kubernetes#1]
Thank you @brakmic for this great tutorial, that's exactly what I needed!

There is a bug however.
Postgres database is created, but Keycloak does not really use it. No Keycloak related tables are created in the database:

```
$ kubectl exec -n hbr-keycloak -it \
> $(kubectl get pods -n hbr-keycloak -o json | jq -r '.items[].metadata.name' | grep postgres) \
> -- psql postgresql://postgres@localhost/keycloak -c 'select name from realm'
ERROR:  relation "realm" does not exist
LINE 1: select name from realm
                         ^
command terminated with exit code 1
```

**Comment by andreygoran** [comment]
After the fix:
```
$ kubectl exec -n hbr-keycloak -it \
> $(kubectl get pods -n hbr-keycloak -o json | jq -r '.items[].metadata.name' | grep postgres) \
> -- psql postgresql://postgres@localhost/keycloak -c 'select name from realm'
  name
--------
 master
(1 row)
```

**Comment by brakmic** [comment]
Many thanks for the patch @andreygoran 

Looks like the same will be needed in the manual variant without helm. 
I suppose the problem is related to variables used in the configuration as they changed between keycloak variants (WildFly & Quarkus).

--EDIT:
I see you changed them as well. Thanks!

**Comment by brakmic** [comment]
Maybe the KC_DB env variable should have been left unchanged, because it's the Keycloak default.

--EDIT:
The same applies to KC_DB_PASSWORD.

**Comment by andreygoran** [comment]
KC_DB is for database vendor and KC_DB_PASSWORD is db password, according to https://www.keycloak.org/server/all-config

**Comment by brakmic** [comment]
> KC_DB is for database vendor and KC_DB_PASSWORD is db password, according to https://www.keycloak.org/server/all-config

Yes, you did everything correctly. My mistake while reading the diff between variants. The other env variable names were written by me. Now I understand where the problem was coming from: Keycloak never had a chance to get any db info because the expected variables (now provided by your patch) didn't exist at all.

Thanks again! 🎉

**Comment by andreygoran** [comment]
Thank you too for this great content!
[https://blog.brakmic.com/keycloak-with-postgresql-on-kubernetes/](https://blog.brakmic.com/keycloak-with-postgresql-on-kubernetes/)

**Comment by brakmic** [comment]
> 

I'm pleased to have been of assistance to you and others. I myself spent days in search of similar content encompassing Helm, Kubernetes, and the Keycloak variant of Quarkus, yet all I could find was a basic rundown of commands. Hence, I penned this guide to save others some time. In addition to this, there is a follow-up article and a [repository](https://github.com/brakmic/Keycloak_with_PostgreSQL-HA_on_Kubernetes) that covers the High Availability (HA) version of Keycloak. However, it doesn't include any manually crafted Helm charts. Instead, I have opted to use the Bitnami packages for Keycloak and PostgreSQL-HA. Included is a [shell script](https://github.com/brakmic/Keycloak_with_PostgreSQL-HA_on_Kubernetes/blob/main/scripts/run_kc_system.sh) that conveniently installs everything in a single execution.
